data/secscan_model/secscan_v4_model.py

```diff
@@ -149,9 +175,32 @@ def needs_reindexing_query(indexer_hash):
         )
 
         for candidate, abt, num_remaining in iterator:
-            # TODO(alecmerdler): Call out to Clair v4 to `/index` the manifest
-            logger.info("Clair v4 client not yet implemented, passing for now")
-            pass
+            manifest = ManifestDataType.for_manifest(candidate, None)
+            layers = registry_model.list_manifest_layers(manifest, self.storage, True)
+
+            logger.debug(
+                "Indexing %s/%s@%s"
+                % (candidate.repository.namespace_user, candidate.repository.name, manifest.digest)
+            )
+
+            (report, state) = self._secscan_api.index(manifest, layers)
+
+            ManifestSecurityStatus.delete().where(
```

Comment:
This is risky; if the create fails after the delete, now we've lost the information until we can reindex. It should either be under a transaction, or you should use `create_or_update`

Reply:
Done using `with db_transaction()`.